rpc/server_interceptors.go

```diff
@@ -0,0 +1,125 @@
+package rpc
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+// UnaryServerTracingInterceptor starts a new Span if Span metadata exists in the context.
+func UnaryServerTracingInterceptor(logger golog.Logger) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		if remoteSpanContext, err := remoteSpanContextFromContext(ctx); err != nil {
+			logger.Warnf("client did not send valid Span metadata in headers, local Spans will not be linked to client", "error", err)
```

Comment:
needs to be warnw as discussed for both but I also think we may not need to log this at all since not all clients will always send spans

Reply:
no worries!